addons/l10n_es_edi_verifactu/tests/test_document.py

```diff
@@ -60,6 +60,105 @@ def test_generation_error(self):
         }
         self.assertRecordValues(invoice, [expected_record_values])
 
+    def test_refund_without_refunded_error(self):
+        "Asserts no error is raised during the generation of the document."
+        invoice = self._create_dummy_invoice(name='INV/2019/00026', invoice_date='2024-12-30')
+        credit_note = invoice._reverse_moves()
+        credit_note.action_post()
+
+        wizard = self.env['account.move.send'].with_context(
+            active_model='account.move',
+            active_ids=credit_note.ids,
+        ).create({
+            'checkbox_send_mail': False,
+            **({'l10n_es_edi_facturae_checkbox_xml': False} if self.env['ir.module.module']._get('l10n_es_edi_facturae').state == 'installed' else {}),
+        })
+        with self._mock_last_document(None), self.assertRaisesRegex(RedirectWarning, r".*There is no Veri\*Factu document for the refunded record\..*"):
+            wizard.action_send_and_print()
+
+        with self._mock_last_document(None):
+            credit_note._l10n_es_edi_verifactu_create_documents()
+
+        errors = ["There is no Veri*Factu document for the refunded record.", "The refund reason is not specified."]
+        expected_record_values = {
+            'l10n_es_edi_verifactu_state': False,
+            'l10n_es_edi_verifactu_warning': self._mock_format_document_generation_errors(errors),
+            'l10n_es_edi_verifactu_warning_level': 'danger',
+        }
+        self.assertRecordValues(credit_note, [expected_record_values])
+
+    def test_substitution_without_errors(self):
```

Comment:
It sounds like we would not encounter any errors in this test
```suggestion
    def test_substitution_without_documents_errors(self):
```